### DISCUSSION TITLE
Visual Studio integration only works when IDE version matches Toolchain version

### DISCUSSION THREAD

**Discussion by ocornut** [microsoft/vcpkg#19224] [Ideas]
**Describe the bug**

I have variety of VS version installed. For this report I'll focus on VS 2015 Update 3 and VS 2019.

When project is setup to use v140 toolchain (VS2015): integration is available in VS2015.
Same project loaded in VS2019: the v140 toolchain is called, but vcpkg integration is not available.
Project needs upgrading to v142 toolchain (VS2019) to have vcpkg integration showing and functional.

This is inconsistent with VS offering access to toolchain selection and makes that toolchain selection basically unusable with vcpkg integration.

Project open in VS2015 IDE:

![image](https://user-images.githubusercontent.com/8225057/121199929-9ca84600-c873-11eb-8106-fec0cfcca129.png)

Same project in VS2019 IDE:

![image](https://user-images.githubusercontent.com/8225057/121200006-ae89e900-c873-11eb-98d3-5ab173e6dc5c.png)
(note that `vcpkg` section is missing)

Same project in VS2019 IDE after upgrading to toolchain v142:

![image](https://user-images.githubusercontent.com/8225057/121200259-e1cc7800-c873-11eb-8e8c-29652c75b417.png)

Here I am showing `vcpkg` section of project properties, as an indicator that vcpkg integration works (it does work in the cases were the vcpkg section appears in that view, it doesn't work in the cases were the vcpkg section doesn't appears).

**Environment**
- OS: Windows
- Compiler: 
Microsoft Visual Studio Community 2015 / Version 14.0.25431.01 Update 3
and
Microsoft Visual Studio Community 2019 / Version 16.9.4 / VisualStudio.16.Release/16.9.4+31205.134

**To Reproduce**
1. ./vcpkg integrate install
2. Create project is VS2015 with v140 toolchain.
3. Load project in VS2019 without a toolchain upgrade.
4. Note that vcpkg integration is inactive.

**Additional context**

This is problematic as it breaks the promise of toolchain selection of VS.

**Comment by PhoebeHui** [comment]
@ocornut, thanks for reporting this issue, this looks not support yet. For this particular case, it looks https://github.com/microsoft/vcpkg/blob/master/scripts/buildsystems/msbuild/vcpkg.props hasn't import. 

@strega-nil-ms, could you please have a look?

**Comment by autoantwort** [comment]
For me this is a bug and it should not be moved to the discussions tab.

**Reply by ocornut** [reply]
This is definitely a bug not a feature, hard to comprehend that a widely advertised feature of VS breaks when using vcpkg.

It’s very problematic for eg projects that needs project files on source control, as things don’t work unless projects are modified to match active IDE version.

Personally this prevents me from recommending my users to use vcpkg.

**Reply by PhoebeHui** [reply]
I also created an issue for this https://github.com/microsoft/vcpkg/issues/19241
